app/assets/javascripts/discourse/tests/acceptance/sidebar-topics-section-test.js

```diff
@@ -407,4 +408,41 @@ acceptance("Sidebar - Topics Section", function (needs) {
       );
     }
   );
+
+  conditionalTest(
+    "adding section link via plugin API",
+    !isLegacyEmber(),
+    async function (assert) {
+      withPluginApi("1.2.0", (api) => {
+        api.addTopicsSectionLink((baseSectionLink) => {
+          return class CustomSectionLink extends baseSectionLink {
```

Comment:
I would really want to see an example with model, and use the model to compute something like name.

Reply:
Updated :+1: